osutil/resolve_path.go

```diff
@@ -0,0 +1,136 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2023 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package osutil
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+func dumbJoin(a, b string) string {
+	if strings.HasSuffix(a, "/") {
+		return a + b
+	} else {
+		return a + "/" + b
+	}
+}
+
+func resolvePathInSysrootRec(sysroot, path string, validInAnySysroot bool, symlinkRecursion int) (string, error) {
```

Comment:
(nitpick) I wonder if `validInAnySysroot` should be called something like `noEscape`, `errorOnEscape` or somesuch to match `ResolvePathNoEscape` ?

Reply:
Changed to `errorOnEscape`.